js/utilities.js

```diff
@@ -1312,31 +1342,81 @@ var ERMrest = (function(module) {
+                    if ((!Array.isArray(ignoredColumns) || ignoredColumns.indexOf(key) == -1) && (keyValues[key] === null || keyValues[key] === undefined)) { 
+                       return false;
                     }
                 }
             }
         }
-
-        var content;
-
-        try {
-            content = module._mustache.render(template, obj);
-        } catch(e) {
-            content = null;
+        return true;
+    };
+    
+    /**
+     * A wrapper for {ERMrest._renderMustacheTemplate}
+     * This function will generate formmatted values from the given data, 
+     * if you don't want the funciton to format the data, make sure to have 
+     * options.formatted = true
+     * 
+     * @param  {ERMrest.Table} table   
+     * @param  {object} data
+     * @param  {string} template 
+     * @param  {string} context 
+     * @param  {Array.<Object>=} options optioanl parameters
+     * @return {string} Returns a string produced as a result of templating using `Mustache`.  
+     */
+    module._renderTemplate = function (template, data, table, context, options) {
+        
+        // to avoid computing data mutliple times, or if we don't want the formatted values
+        if (options === undefined || !options.formatted) {
+            data = module._getFormattedKeyValues(table.columns, context, data);
         }
-
-        return content;
+        
+        // render the template using Mustache
+        return module._renderMustacheTemplate(template, data, options);
+    };
+    
+    /**
+     * A wrapper for {ERMrest._validateMustacheTemplate} 
+     * it will take care of adding formmatted and unformatted values.
+     * options.formmatted=true: to avoid formatting key values
+     * options.ignoredColumns: list of columns that you want validator to ignore
+     * 
+     * @param  {ERMrest.Table} table   
+     * @param  {object} data
+     * @param  {string} template 
+     * @param  {string} context 
+     * @param  {Array.<string>=} ignoredColumns the columns that should be ignored (optional)
+     * @return {boolean} True if the template is valid.
+     */
+    module._validateTemplate = function (template, data, table, context, options) {
+        
+        var ignoredColumns;
+        if (typeof options !== undefined && Array.isArray(options.ignoredColumns)) {
+            ignoredColumns = options.ignoredColumns;
+        }
+        
+        // to avoid computing data multiple times, or if we don't want the formatted values
+        if (options === undefined || !options.formatted) {
+            // make sure to add formmatted columns too.
+            if (ignoredColumns !== undefined) {
+                ignoredColumns.forEach(function (col) {
+                    ignoredColumns.push("_" + col);
+                });
+            }
+            
+            data = module._getFormattedKeyValues(table.columns, context, data);
+        }
+        
+        // render the template using Mustache
```

Comment:
Seems like this comment is incorrect. My understanding is that you are validating whether or not it is a valid template and then passing that check off to verify if it's a valid mustache template.